simpledotorg/simple-server
# Online patient lookup API

**Story card:** [ch3257](https://app.clubhouse.io/simpledotorg/story/3257/add-a-new-api-for-looking-up-patients-by-their-identifier)

## Because
This implements the server side of the online patient lookup feature.

- [ADR](https://github.com/simpledotorg/simple-server/blob/master/doc/arch/015-online-patient-lookup.md)
- [PRD](https://docs.google.com/document/d/1q6cppByQULfh3_mMXC4BJpiNN9Uc_awA6rreeEtUBaM/edit#)

## This addresses

- API implementation, including schema, transformation
- Access restriction to state level data
- Audit logging
- Rate limiting

## Test instructions

- Make requests to `/api/v4/patients/{identifier}`
- Ensure it works for a single patient, and then multiple patients with the same identifier
- Test rate limiting end to end
- Test that audit logs are written to file in a separate file
